carrow.go

```diff
@@ -40,20 +37,22 @@ type Field struct {
 }
 
 // NewField returns a new Field
-func NewField(name string, dtype DType) (*Field, error) {
+func NewField(name string, dtype C.int) (*Field, error) {
```

Comment:
Why not stick to `DType`? This is exposed to the user and should not "leak" implementation details.